In [1]:
import pandas as pd
import numpy as np
import math

# Read in the data
gpt_df = pd.read_csv('./Data/free_text_input/Results/GPT4_top_prediction_RE.csv')
# split columns into multiple columns
# gpt_df['pt_id'] = gpt_df['file_path']
gpt_df['PMID'] = gpt_df['file_path'].apply(lambda x: x.split('_')[0])

In [2]:
probe_info_df = pd.read_csv('./Data/HPO_input/Original_data/probe_info', sep = '\t', header=None)
probe_info_df.columns = ['dataset','pt_id','dx_gene']
probe_info_df['PMID'] = probe_info_df['pt_id'].apply(lambda x: str(x).split('_')[0])
probe_info_df = probe_info_df[['dataset','PMID']].drop_duplicates()

In [3]:
df = gpt_df.merge(probe_info_df, how='left')
df['dataset'] = df['dataset'].fillna('Others')

In [43]:
df[df['dataset'] == 'AJHG']

file_path dx_gene  predict_made_in_top_5  predict_correct_in_top_5   
0    PMID28686853_0   WDR26                      0                         0  \
1    PMID28686853_1   WDR26                      0                         0   
2    PMID28686853_2   WDR26                      0                         0   
3    PMID28686853_3   WDR26                      0                         0   
4    PMID28686853_4   WDR26                      0                         0   
..              ...     ...                    ...                       ...   
91   PMID29100085_7   DHX30                      0                         0   
92   PMID29100085_8   DHX30                      0                         0   
93   PMID29100085_9   DHX30                      0                         0   
94  PMID29100085_10   DHX30                      0                         0   
95  PMID29100085_11   DHX30                      0                         0   

    predict_made_in_top_10  predict_correct_in_top_10  predict_made_in_top_50   
0                        1                          0                       1  \
1                        1                          0                       1   
2                        1                          0                       1   
3                        1                          0                       1   
4                        1                          0                       1   
..                     ...                        ...                     ...   
91                       1                          0                       0   
92                       1                          0                       1   
93                       1                          0                       0   
94                       1                          0                       1   
95                       1                          0                       1   

    predict_correct_in_top_50          PMID dataset  
0                           0  PMID28686853    AJHG  
1                           0  PMID28686853    AJHG  
2                           0  PMID28686853    AJHG  
3                           0  PMID28686853    AJHG  
4                           0  PMID28686853    AJHG  
..                        ...           ...     ...  
91                          0  PMID29100085    AJHG  
92                          0  PMID29100085    AJHG  
93                          0  PMID29100085    AJHG  
94                          0  PMID29100085    AJHG  
95                          0  PMID29100085    AJHG  

[72 rows x 10 columns]

In [4]:
# format float to 2 decimal places
w = df.shape[0]
x = df[['predict_correct_in_top_5']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_10']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_50']].values.sum()
y = df[['predict_made_in_top_50']].values.sum()
print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))


125 out of 125 predictions made in top 5 (100.0%)
13 out of 125 predictions correct in top 5 (10.4%)
125 out of 125 predictions made in top 10 (100.0%)
12 out of 125 predictions correct in top 10 (9.6%)
79 out of 125 predictions made in top 50 (63.2%)
12 out of 79 predictions correct in top 50 (15.190000000000001%)


In [52]:
# This table is generated using ChatGPT-3

In [5]:
# subset df by dataset
dataset_list = df['dataset'].unique()
for dataset in dataset_list:
    print(dataset)
    df_subset = df[df['dataset']==dataset]
    w = df_subset.shape[0]
    x = df_subset[['predict_correct_in_top_5']].values.sum()
    y = df_subset[['predict_made_in_top_5']].values.sum()
    print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df_subset[['predict_correct_in_top_10']].values.sum()
    y = df_subset[['predict_made_in_top_10']].values.sum()
    print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df_subset[['predict_correct_in_top_50']].values.sum()
    y = df_subset[['predict_made_in_top_50']].values.sum()
    print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))
    print('')

AJHG
72 out of 72 predictions made in top 5 (100.0%)
0 out of 72 predictions correct in top 5 (0.0%)
72 out of 72 predictions made in top 10 (100.0%)
0 out of 72 predictions correct in top 10 (0.0%)
45 out of 72 predictions made in top 50 (62.5%)
0 out of 45 predictions correct in top 50 (0.0%)

CSH
49 out of 49 predictions made in top 5 (100.0%)
12 out of 49 predictions correct in top 5 (24.490000000000002%)
49 out of 49 predictions made in top 10 (100.0%)
11 out of 49 predictions correct in top 10 (22.45%)
31 out of 49 predictions made in top 50 (63.27%)
11 out of 31 predictions correct in top 50 (35.480000000000004%)

Others
4 out of 4 predictions made in top 5 (100.0%)
1 out of 4 predictions correct in top 5 (25.0%)
4 out of 4 predictions made in top 10 (100.0%)
1 out of 4 predictions correct in top 10 (25.0%)
3 out of 4 predictions made in top 50 (75.0%)
1 out of 3 predictions correct in top 50 (33.33%)



| Dataset | Predictions in Top 5 | Correct Predictions in Top 5 | Predictions in Top 10 | Correct Predictions in Top 10 | Predictions in Top 50 | Correct Predictions in Top 50 |
|---------|---------------------|------------------------------|-----------------------|--------------------------------|------------------------|-------------------------------|
| AJHG    | 72 out of 72 (100%)  | 0 out of 72 (0.0%)            | 72 out of 72 (100%)    | 0 out of 72 (0.0%)             | 45 out of 72 (62.5%)   | 0 out of 45 (0.0%)            |
| CSH     | 49 out of 49 (100%)  | 12 out of 49 (24.49%)         | 49 out of 49 (100%)    | 11 out of 49 (22.45%)          | 31 out of 49 (63.27%)  | 11 out of 31 (35.48%)         |
| Others  | 4 out of 4 (100%)    | 1 out of 4 (25.0%)            | 4 out of 4 (100%)      | 1 out of 4 (25.0%)             | 3 out of 4 (75.0%)     | 1 out of 3 (33.33%)           |
